api.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { expect } from 'chai';
+import dotenv from 'dotenv';
+import { VERSION } from './version';
+dotenv.config({ path: '.env' });
+
+import WhispirClient from './api';
+
+let client;
+const workspaceId = process.env.WORKSPACE_ID;
+const messageId = '55294920E647F399';
+
+beforeEach(() => {
+})
+
+describe("Whispir Client", () => {
+    it("should accept username and password", async () => {
+        client = WhispirClient({
+            host: 'https://stage-ap-southeast-2.whispirdev.com/api',
```

Comment:
I think we probably want to setup a mock server for these tests, to decouple these tests from any implementation issues.

We can use Stoplight's [Prism](https://stoplight.io/open-source/prism) mock server here - long term, we can setup a Docker image as Stripe does, but for now it's perfectly fine to just start up Prism from the CLI, like so:

```bash
> npx @stoplight/prism-cli mock https://raw.githubusercontent.com/whispir/openapi/main/openapi.yaml
[12:00:55 pm] › [CLI] ▶  start     Prism is listening on http://127.0.0.1:4010
```

Then, we can use the mock server as the host:
```
host: 'http://127.0.0.1:4010',
```

and test that the SDK get's the expected response.

ps. test curl command:
```bash
> curl --request POST \
  --url http://127.0.0.1:4010/auth \
  --header 'Accept: application/vnd.whispir.auth-v1+json' \
  --header 'Authorization: Basic XXXXX==' \
  --header 'Content-Type: application/vnd.whispir.auth-v1+json' \
  --header 'Api-Key: XXXXX' \
  --header 'X-Api-Key: XXXXX'
{"token":"your-generated-token-string","link":[]}
```

Reply:
I have to update the `openapi.yaml` spec here: https://github.com/whispir/openapi/pull/154